pkg/controllers/selection/controller.go

```diff
@@ -142,11 +142,11 @@ func validateAffinity(p *v1.Pod) (errs error) {
 	if p.Spec.Affinity == nil {
 		return nil
 	}
-	if pod.HasPodAffinity(p) {
-		errs = multierr.Append(errs, fmt.Errorf("pod affinity is not supported"))
+	if pod.HasRequiredPodAffinity(p) {
+		errs = multierr.Append(errs, fmt.Errorf("strong pod affinity is not supported"))
```

Comment:
I'm not sure if this message is clear enough.
another alternative could be "pod affinity required during scheduling is not supported"